### ISSUE TITLE
schema-reader Error: Database file at path

### ISSUE THREAD

**Issue by juliangums** [VemtoOrg/vemto2-issues#72]
Weirdly I selected MySQL, not SQLite:

`schema-reader Error: Database file at path [phar:///Applications/Vemto.app/Contents/Resources/static/database/database.sqlite] does not exist. Ensure this is an absolute path to the database. (Connection: sqlite, SQL: delete from "cache" where "key" = laravel_cache_spatie.permission.cache)`

**Comment by TiagoSilvaPereira** [comment]
@juliangums please check the Cache driver of your application (It is probably using the sqlite connection)

**Comment by TiagoSilvaPereira** [comment]
@juliangums please try changing your CACHE driver to file for now... I just noticed I'll need to change the cache database path when running from Vemto. Thanks for reporting!

**Comment by juliangums** [comment]
@TiagoSilvaPereira no it is using file: `CACHE_DRIVER=file`. I don't think I have anything that uses SQLite

**Comment by TiagoSilvaPereira** [comment]
@juliangums is this an existing project or a new one? Could you please try to reproduce it?

**Comment by juliangums** [comment]
Existing

**Comment by juliangums** [comment]
If I delete the migration for Spatie Permissions I can import it. So something in this file that the package generates breaks it: 
```php
<?php

use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

return new class extends Migration
{
    /**
     * Run the migrations.
     */
    public function up(): void
    {
        $teams = config('permission.teams');
        $tableNames = config('permission.table_names');
        $columnNames = config('permission.column_names');
        $pivotRole = $columnNames['role_pivot_key'] ?? 'role_id';
        $pivotPermission = $columnNames['permission_pivot_key'] ?? 'permission_id';

        if (empty($tableNames)) {
            throw new \Exception('Error: config/permission.php not loaded. Run [php artisan config:clear] and try again.');
        }
        if ($teams && empty($columnNames['team_foreign_key'] ?? null)) {
            throw new \Exception('Error: team_foreign_key on config/permission.php not loaded. Run [php artisan config:clear] and try again.');
        }

        Schema::create($tableNames['permissions'], function (Blueprint $table) {
            $table->bigIncrements('id'); // permission id
            $table->string('name');       // For MySQL 8.0 use string('name', 125);
            $table->string('guard_name'); // For MySQL 8.0 use string('guard_name', 125);
            $table->timestamps();

            $table->unique(['name', 'guard_name']);
        });

        Schema::create($tableNames['roles'], function (Blueprint $table) use ($teams, $columnNames) {
            $table->bigIncrements('id'); // role id
            if ($teams || config('permission.testing')) { // permission.testing is a fix for sqlite testing
                $table->unsignedBigInteger($columnNames['team_foreign_key'])->nullable();
                $table->index($columnNames['team_foreign_key'], 'roles_team_foreign_key_index');
            }
            $table->string('name');       // For MySQL 8.0 use string('name', 125);
            $table->string('guard_name'); // For MySQL 8.0 use string('guard_name', 125);
            $table->timestamps();
            if ($teams || config('permission.testing')) {
                $table->unique([$columnNames['team_foreign_key'], 'name', 'guard_name']);
            } else {
                $table->unique(['name', 'guard_name']);
            }
        });

        Schema::create($tableNames['model_has_permissions'], function (Blueprint $table) use ($tableNames, $columnNames, $pivotPermission, $teams) {
            $table->unsignedBigInteger($pivotPermission);

            $table->string('model_type');
            $table->unsignedBigInteger($columnNames['model_morph_key']);
            $table->index([$columnNames['model_morph_key'], 'model_type'], 'model_has_permissions_model_id_model_type_index');

            $table->foreign($pivotPermission)
                ->references('id') // permission id
                ->on($tableNames['permissions'])
                ->onDelete('cascade');
            if ($teams) {
                $table->unsignedBigInteger($columnNames['team_foreign_key']);
                $table->index($columnNames['team_foreign_key'], 'model_has_permissions_team_foreign_key_index');

                $table->primary([$columnNames['team_foreign_key'], $pivotPermission, $columnNames['model_morph_key'], 'model_type'],
                    'model_has_permissions_permission_model_type_primary');
            } else {
                $table->primary([$pivotPermission, $columnNames['model_morph_key'], 'model_type'],
                    'model_has_permissions_permission_model_type_primary');
            }

        });

        Schema::create($tableNames['model_has_roles'], function (Blueprint $table) use ($tableNames, $columnNames, $pivotRole, $teams) {
            $table->unsignedBigInteger($pivotRole);

            $table->string('model_type');
            $table->unsignedBigInteger($columnNames['model_morph_key']);
            $table->index([$columnNames['model_morph_key'], 'model_type'], 'model_has_roles_model_id_model_type_index');

            $table->foreign($pivotRole)
                ->references('id') // role id
                ->on($tableNames['roles'])
                ->onDelete('cascade');
            if ($teams) {
                $table->unsignedBigInteger($columnNames['team_foreign_key']);
                $table->index($columnNames['team_foreign_key'], 'model_has_roles_team_foreign_key_index');

                $table->primary([$columnNames['team_foreign_key'], $pivotRole, $columnNames['model_morph_key'], 'model_type'],
                    'model_has_roles_role_model_type_primary');
            } else {
                $table->primary([$pivotRole, $columnNames['model_morph_key'], 'model_type'],
                    'model_has_roles_role_model_type_primary');
            }
        });

        Schema::create($tableNames['role_has_permissions'], function (Blueprint $table) use ($tableNames, $pivotRole, $pivotPermission) {
            $table->unsignedBigInteger($pivotPermission);
            $table->unsignedBigInteger($pivotRole);

            $table->foreign($pivotPermission)
                ->references('id') // permission id
                ->on($tableNames['permissions'])
                ->onDelete('cascade');

            $table->foreign($pivotRole)
                ->references('id') // role id
                ->on($tableNames['roles'])
                ->onDelete('cascade');

            $table->primary([$pivotPermission, $pivotRole], 'role_has_permissions_permission_id_role_id_primary');
        });

        app('cache')
            ->store(config('permission.cache.store') != 'default' ? config('permission.cache.store') : null)
            ->forget(config('permission.cache.key'));
    }

    /**
     * Reverse the migrations.
     */
    public function down(): void
    {
        $tableNames = config('permission.table_names');

        if (empty($tableNames)) {
            throw new \Exception('Error: config/permission.php not found and defaults could not be merged. Please publish the package configuration before proceeding, or drop the tables manually.');
        }

        Schema::drop($tableNames['role_has_permissions']);
        Schema::drop($tableNames['model_has_roles']);
        Schema::drop($tableNames['model_has_permissions']);
        Schema::drop($tableNames['roles']);
        Schema::drop($tableNames['permissions']);
    }
};
```

**Comment by juliangums** [comment]
If I comment this out it works:

```php
app('cache')
            ->store(config('permission.cache.store') != 'default' ? config('permission.cache.store') : null)
            ->forget(config('permission.cache.key'));
```

**Comment by TiagoSilvaPereira** [comment]
@juliangums hmm maybe Spatie permissions trying to use sqlite as the cache engine? I'll try to reproduce it

**Comment by juliangums** [comment]
It's fine if I run `php artisan migrate:fresh --seed`.
And my config and everything point to file or default and default points to file. Perhaps Vemto somehow uses SQLite as a default instead? SQLite isn't even configured on my project.

**Comment by TiagoSilvaPereira** [comment]
Hi @juliangums I tried to reproduce it on a new project with Spatie permissions and Database Reader (MySQL) but without success. It managed to read the project, including the Spatie migrations. Could you please check if this happens with a new project with Spatie Permissions installed?

![image](https://github.com/TiagoSilvaPereira/vemto2-issues/assets/11933789/3c7e31b9-97f3-42a2-9f4e-f782bca62de1)

**Comment by TiagoSilvaPereira** [comment]
Also, please check if the spatie configuration of your project (permission.php) is using your default cache driver (although I need to fix Vemto to consider whatever you add to this configuration path)

![image](https://github.com/TiagoSilvaPereira/vemto2-issues/assets/11933789/cc6c39d6-a470-43dc-9dca-427d4f6e35c5)

**Comment by TiagoSilvaPereira** [comment]
Hmmm I had a similar problem here when running:

```
php artisan config:clear
```

And after running:

```
php artisan config:cache
```

The issue is gone. 

Maybe it is a similar issue (I guess that the Seeder runs this command under the hoods)? 

I think I'll need to force the config cache before reading the schema migrations (Not sure if this is a good practice)

**Comment by juliangums** [comment]
Yes, that's what it is. Works when I cache it first. Don't think it's bad practice to cache the config first. It does lead to another unrelated error though that I don't get if I just comment the line out in the migration.

**Comment by juliangums** [comment]
it is specifically `->forget(config('permission.cache.key'))` causing it. It tries to forget this but cannot find it cached.

**Comment by TiagoSilvaPereira** [comment]
Awesome, I'll change the schema reader to cache settings before running the migrations

**Comment by juliangums** [comment]
The issue is that this threw another error for me related to my Filament custom icon set.

**Comment by TiagoSilvaPereira** [comment]
@juliangums I think I found a way to fix this issue (Vemto as executing the Laravel application from the wrong path). I'll compile 2.0.8 (probably today) with this bug fix

**Comment by TiagoSilvaPereira** [comment]
@juliangums fixed it on Vemto 2.0.8, and now the database reader is booting the application correctly. I'm closing this issue for now, but feel free to reopen or open another issue if you find other issues

**Comment by juliangums** [comment]
Works well now.
